core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/upgrade/rds/UpgradeRdsHandler.java

```diff
@@ -51,15 +66,47 @@ public Selectable doAccept(HandlerEvent<UpgradeRdsUpgradeDatabaseServerRequest>
         UpgradeRdsUpgradeDatabaseServerRequest request = event.getData();
         Long stackId = request.getResourceId();
         LOGGER.info("Starting RDS database upgrade...");
+        StackEvent result;
+        StackDto stackDto = stackDtoService.getById(stackId);
         try {
-            StackDto stackDto = stackDtoService.getById(stackId);
             if (embeddedDatabaseService.isAttachedDiskForEmbeddedDatabaseCreated(stackDto)) {
-                LOGGER.debug("Starting embedded database upgrade...");
-                rdsUpgradeOrchestratorService.upgradeEmbeddedDatabase(stackDto);
+                result = upgradeEmbeddedDatabase(request, stackId, stackDto);
+            } else {
+                result = upgradeExternalDatabase(request, stackId, stackDto);
```

Comment:
this is not in sync with the condition. if the embedded db is on the root disk we would start an external db upgrade?

Reply:
Good point, I am not really sure, what should happen with embedded db's on the root disk, I assume there should be a validation not to let them in, @keyki @schfeca75 could you pls comment?

Reply:
Sorted out the 3 different conditions